spec/unit/hathifiles_database_verifier_spec.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+require "verifier/hathifiles_database_verifier"
+
+module PostZephirProcessing
+  RSpec.describe(HathifilesDatabaseVerifier) do
+    around(:each) do |example|
+      with_test_environment do
+        ClimateControl.modify(HATHIFILE_ARCHIVE: fixture("hathifile_archive")) do
+          example.run
+        end
+      end
+    end
+
+    let(:verifier) { described_class.new }
+    let(:test_update_file) { "hathi_upd_20241202.txt.gz" }
+    let(:test_update_fixture) { fixture(File.join("hathifile_archive", test_update_file)) }
+    let(:test_full_file) { "hathi_full_20241201.txt.gz" }
+
+    def delete_hf_logs
+      Services[:database][:hf_log].delete
+    end
+
+    # Temporarily add `hathifile` to `hf_log` with the current timestamp.
+    def with_fake_hf_log_entry(hathifile:)
+      delete_hf_logs
+      Services[:database][:hf_log].insert(hathifile: hathifile)
+      begin
+        yield
+      ensure
+        delete_hf_logs
+      end
+    end
+
+    # Temporarily add `htid` to `hf` with reasonable (and irrelevant) defaults.
+    def with_fake_hf_entries(htids:)
+      Services[:database][:hf].where(htid: htids).delete
+      htids.each { |htid| Services[:database][:hf].insert(htid: htid) }
+      begin
+        yield
+      ensure
+        Services[:database][:hf].where(htid: htids).delete
+      end
+    end
+
+    # Copies one of our fixtures into @tmpdir and renames it.
+    # Could just use an additional fixture if we're not worried about the proliferation
+    # of them.
+    def with_fake_full_hathifile
+      ClimateControl.modify(HATHIFILE_ARCHIVE: @tmpdir) do
+        FileUtils.cp(test_update_fixture, File.join(@tmpdir, test_full_file))
+        yield
+      end
+    end
+
+    describe ".has_log?" do
+      context "with corresponding hf_log" do
+        it "returns `true`" do
+          with_fake_hf_log_entry(hathifile: "hathi_upd_20241202.txt.gz") do
+            expect(described_class.has_log?(hathifile: test_update_fixture)).to eq(true)
```

Comment:
`be_true` / `be_false`?

Reply:
Apparently those got renamed `be_truthy` and `be_falsey` (yuck) so I am shifting to the fairly strict `be true` and `be false`